sdk/core/dma/dma_compartment.cc

```diff
@@ -0,0 +1,109 @@
+#pragma once
+
+#define MALLOC_QUOTA 0x100000
+
+#include <cheri.hh>
+#include <compartment-macros.h>
+#include <utils.hh>
+#include <thread_pool.h>
+
+#if __has_include(<platform-dma.hh>)
+#	include <platform-dma.hh>
+#endif
+
+// Import some useful things from the CHERI namespace.
+using namespace CHERI;
+
+using thread_pool::async;
+/**
+ *  creating a global heap below.
+ *  todo: we assume that this heap capability can serve 
+ *  for different compartments, irrespective of 
+ *  the origin because this is a driver code?
+*/
+DECLARE_AND_DEFINE_ALLOCATOR_CAPABILITY(dmaDriverHeap, 8144);
```

Comment:
If you’re using a named heap capability, why did you provide a large quota to the malloc capability?